llmc/zero.cuh

```diff
@@ -593,5 +593,17 @@ float multi_gpu_cpu_float_sum(float value, MultiGpuConfig* config) {
 #endif
 }
 
+template<int N>
+void zero2_accumulate_grad(floatX* const (&dst)[N], floatX* const (&src)[N], const size_t (&nelem)[N], int layer, unsigned seed, cudaStream_t stream) {
+    for(int i = 0; i < N; ++i) {
+        size_t n = nelem[i] / multi_gpu_config.num_processes;
+        vector_add<<<CEIL_DIV(n, 512), 512, 0, stream>>>(dst[i] + layer * n,
+                                                              src[i] + multi_gpu_config.process_rank * n,
+                                                              n, seed + i);
+        cudaCheck(cudaGetLastError());
+        cudaCheck(cudaMemset(src[i], 0, nelem[i] * sizeof(floatX)));
```

Comment:
could be fused into the vector_add - these are tiny kernel calls in the same stream, so the idle time between might be significant (i.e. noticeable in Nsight Systems but not Nsight Compute)

Reply:
If we move zeroing (pun intended) inside, I think we should then move the `vector_add` function from `cuda_utils.h`  to `zero.h`.